### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by Shevtsooov**
[DEMO LINK](https://Shevtsooov.github.io/react_clock/)

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -1,39 +1,56 @@
...
+    document.addEventListener('contextmenu', (event) => {
+      event.preventDefault();
+
+      this.setState({ hasClock: false });
+    });
+  }
+
+  componentWillUnmount() {
+    window.clearInterval(this.timerId);
+  }
```
Comment (silchyn) [inline src/App.tsx:41]: event listeners should also be removed

File: src/components/Clocks/Clock.tsx [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    window.clearInterval(this.timerUpdate);
+  }
+
+  render() {
+    const { today } = this.state;
+
+    return (
+      <div className="Clock">
+        <strong className="Clock__name">
+          {this.props.clockName}
```
Comment (silchyn) [inline src/components/Clocks/Clock.tsx:47]: destructure `props`

**Review by silchyn — CHANGES_REQUESTED** [review]

File: src/App.tsx [hunk 3]
```diff
@@ -1,39 +1,62 @@
...
+
+  rightClickListener = (e: MouseEvent) => {
+    e.preventDefault();
+
+    this.setState({ hasClock: false });
+  };
+
+  leftClickListener = () => {
+    this.setState({ hasClock: true });
+  };
```
Comment (naziamoff) [inline src/App.tsx:47]:
```suggestion
  handleRightClick = (e: MouseEvent) => {
    e.preventDefault();

    this.setState({ hasClock: false });
  };

  handleLeftClick = () => {
    this.setState({ hasClock: true });
  };
```

**Review by naziamoff — APPROVED** [review]
looks good 👍

**Pull request closed without merging**